orb-ui/src/sound/mod.rs

```diff
@@ -314,10 +329,83 @@ impl Player for Jetson {
         }
     }
+
+    /// Finds the sound file path for the given sound type and sends it to the
+    /// sound player only if the sink is empty
+    fn try_queue(&self, sound_type: Type) -> Result<bool> {
+        return if let Ok(sink) = self.sink.lock() {
+            if sink.empty() {
+                if let Some(sound_file) = self.sound_files.get(&sound_type) {
+                    if let Some(sound_file) = sound_file.value() {
+                        if let Ok(mut tx_queue) = self.queue_file.lock() {
+                            tx_queue
+                                .try_send(sound_file.clone())
+                                .wrap_err("Failed to queue sound")?;
+                            Ok(true)
+                        } else {
+                            Err(eyre!("Failed to lock queue"))
+                        }
+                    } else {
+                        Err(eyre!(
+                            "Sound file {:?} doesn't have a known file path",
+                            sound_type
+                        ))
+                    }
+                } else {
+                    Err(eyre!("Sound file not found: {:?}", sound_type))
+                }
+            } else {
+                Ok(false)
+            }
+        } else {
+            Err(eyre!("Failed to lock sink"))
+        };
+    }
+
+    fn set_volume(&self, volume_percent: u64) {
+        if let Ok(sink) = self.sink.lock() {
+            sink.set_volume((volume_percent as f64 / 100_f64) as f32);
+        } else {
+            tracing::error!("Failed to lock sink to set sound volume");
+        }
+    }
+
+    fn set_language(&self, language: Option<&str>) {
+        self.sound_files.clear();
+        let language = language.map(ToOwned::to_owned);
+
+        if let Err(e) =
+            Voice::load_sound_files(SOUNDS_DIR, &self.sound_files, language.as_deref())
+        {
+            tracing::error!("Failed to load voice sound files: {:?}", e);
+        }
+
+        if let Err(e) =
+            Melody::load_sound_files(SOUNDS_DIR, &self.sound_files, language.as_deref())
+        {
+            tracing::error!("Failed to load melody sound files: {:?}", e);
+        }
+    }
 }
 
 fn load_filepaths(dir: &str, sound: &str, language: Option<&str>) -> Option<String> {
-    let path = format!("{}/{}{}.wav", dir, sound, language.unwrap_or(""));
+    // if a `language` is passed and the sound is a voice, make sure we append the
+    // localized language to the file name
+    // e.g. voice_server_error__es-ES.wav
+    let has_extension = language.is_some()
+        && !language.unwrap().contains("en-")
+        && sound.contains("voice_");
+    let lang_extension = if has_extension {
+        if let Some(language) = language {
+            format!("__{}", language)
+        } else {
+            "".to_string()
+        }
+    } else {
+        "".to_string()
+    };
+
+    let path = format!("{}/{}{}.wav", dir, sound, lang_extension);
```

Comment:
generally, you should be using `std::path::Path` and `.join()` instead of manually formatting your filepaths as strings.

Reply:
now using PathBuf, which is cross platform, instead of these hardcoded filepaths